pkg/controllers/networking/pod_controller.go

```diff
@@ -498,7 +516,26 @@ func squashIPSliceToSubnets(ips []*types.IP) (ret []string) {
 }
 
 // SetupWithManager sets up the controller with the Manager.
-func (r *PodReconciler) SetupWithManager(mgr ctrl.Manager) error {
+func (r *PodReconciler) SetupWithManager(mgr ctrl.Manager) (err error) {
+	// init node indexer for networks
+	if err = mgr.GetFieldIndexer().IndexField(context.TODO(), &networkingv1.Network{}, indexerFieldNode, func(obj client.Object) []string {
+		network, ok := obj.(*networkingv1.Network)
+		if !ok {
+			return nil
+		}
+
+		switch network.Spec.Type {
```

Comment:
Use `func GetNetworkType(networkObj *Network) NetworkType` here?  `network.Spec.Type` might be empty.